storage/storage_reference.go

```diff
@@ -0,0 +1,90 @@
+package storage
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/containers/image/types"
+	"github.com/containers/storage/storage"
+	"github.com/docker/docker/reference"
+)
+
+// A storageReference holds a name and/or an ID, which is a 32-byte value
+// hex-encoded into a 64-character string.
+type storageReference struct {
+	store     storage.Store
+	transport *storageTransport
+	reference string
+	id        string
+}
+
+func newReference(store storage.Store, transport *storageTransport, reference, id string) *storageReference {
+	return &storageReference{
+		store:     store,
+		transport: transport,
+		reference: reference,
+		id:        id,
+	}
+}
+
+// Resolve the reference's name to an image ID in the storage library if
+// there's already one present with the same name or ID.
+func (s *storageReference) ID() string {
+	if s.id == "" {
+		image, err := s.store.GetImage(s.reference)
+		if image != nil && err == nil {
+			s.id = image.ID
+		}
+	}
+	return s.id
+}
+
+func (s *storageReference) Transport() types.ImageTransport {
+	s.ID()
+	return s.transport
+}
+
+func (s *storageReference) DockerReference() reference.Named {
+	s.ID()
```

Comment:
what are those calls needed for? just to populate the image id? if so, here and in Transport() are those calls needed?